CAPI/API/src/DebugAPI.cpp

```diff
@@ -88,9 +88,12 @@ int ButcherDebugAPI::GetFrameCount() const
 
 std::future<bool> HumanDebugAPI::Move(int64_t timeInMilliseconds, double angleInRadian)
 {
-    logger->info("Move: timeInMilliseconds = {}, angleInRadian = {}, at {}ms", timeInMilliseconds, angleInRadian, Time::TimeSinceStart(startPoint));
+    logger->info("Move: timeInMilliseconds = {}, angleInRadian = {}, called at {}ms", timeInMilliseconds, angleInRadian, Time::TimeSinceStart(startPoint));
     return std::async(std::launch::async, [&]()
```

Comment:
```suggestion
    return std::async(std::launch::async, [=, this]()
```


这段代码有**严重**的问题。
lambda 表达式语法：

```ANTLR
Lambda
  : '[' CaptureList ']' ('('((Param (',' Param)+)|)')')? Specs? (-> Type)? '{' Statement+ '}'
  ;
CaptureList
  : /*  Empty */
  | Captured (',' Captured)?
  ;
Captured
  : ('&' | '=')  /* 按引用或值捕获全部 automatic 变量、参数以及 `this` 指针 */
  : ('&')? Indentifier ('=' Identifier)?
  : "this"       /* 捕获 `this` 指针以访问成员变量 */
  : '*' "this"   /* 复制捕获当前对象以访问拷贝对象的成员变量 */
  ;
```

如果用 `&` 的话，则按引用捕获全部 automatic 变量、参数以及 `this` 指针。在本例中，只是为了访问 `logic` 以及 `logger`，实质上只是用到了捕获的 `this` 指针。而由于我们的 lambda 表达式是在函数退出后运行的，因此引用捕获局部变量的话，在函数退出后访问 `timeInMilliseconds`、`angleInRadian` 的引用会产生未定义行为。因此应该改为 `[=, this]`。

之后的所有的代码（`MoveDown`、`Move`、`PickProp` 等）同理。

Reply:
`explicit by-copy capture of ‘this’ redundant with by-copy capture default`
为什么会有这样一个warning啊（

Reply:
我感觉`[=]`和`[=, this]`是等价的？

Reply:
哦，我傻了，确实写个 `[=]` 就行了

Reply:
`[=, this]` 这个貌似是 C++20 才允许，可以看这个：<https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0409r2.html>

Reply:
emmm查了一下，似乎C++17默认的`[=]`捕获中`this`是捕获指针的，也可以写成`[=, *this]`来复制整个对象；而C++20之后才要求明确写明`this`或`*this`